### PR TITLE
Notify service

### LINKED ISSUE #427: Create skeleton service for notify service
A simple go service skeleton for the service `notify`.

- Create it in the `cmd` subfolder
- It needs to recieve messages from the MQ, so a loop over the MQ is the base

### PR DESCRIPTION
**Pull request by jbygdell**
Some unit tests, not fully for smtp though
No integration tests

Closes #427

### CODE REVIEW

**Review by norling — APPROVED** [review]
Structure seems fine, but I'd like some description comments! :)

**Review by aaperis — APPROVED** [review]
Looks good  :-).  I agree with previous remark on comments. The `common` package could be quite useful for further shortening the code in the future. In its current form though, it would be nice if it included some short note that code is repeated.

File: internal/common/common.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package common
+
+import (
+	"github.com/xeipuuv/gojsonschema"
+)
+
+func ValidateJSON(reference string, body []byte) (*gojsonschema.Result, error) {
```
[the remark this answers is not in the document]
Comment (jbygdell) [reply internal/common/common.go:7]: The function in the broker package will be moved into common later, then there will be no confusion anymore.
I don't wan't to put it in the PR as it is better to have that as a separate one.

File: cmd/notify/notify.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+				continue
+			}
+			user := getUser(conf.Broker.Queue, d.Body)
+			if user == "" {
+				log.Errorln("No user in message, skipping")
+
+				continue
+			}
+
+			if err := sendEmail(conf.Notify, "THIS SHOULD TAKE A TEMPLATE", user, setSubject(conf.Broker.Queue)); err != nil {
```
Comment (blankdots) [inline cmd/notify/notify.go:66]: `"THIS SHOULD TAKE A TEMPLATE"` we leave like this or we configure an external file to read from?
should/can we use go templates for this ?

or we solve this in a different PR where we can add multiple templates depending on what happened per queue ?

Comment (jbygdell) [reply cmd/notify/notify.go:66]: The idea was to have a template file depending on  the reason for the email being sent, to be solved down the road in separate PRs.

Comment (blankdots) [reply cmd/notify/notify.go:66]: i am ok with this.

**Review by blankdots — APPROVED** [review]

**Comment by viklund** [comment]
I think we can merge this now. It's mostly relevant for Fi and Se at the moment.

**Pull request merged**